code/game/objects/structures/grille.dm

```diff
@@ -78,10 +78,15 @@
 		shock(user, 70)
 		shockcooldown = world.time + my_shockcooldown
 
-/obj/structure/grille/attack_animal(mob/user)
+/obj/structure/grille/attack_animal(mob/living/simple_animal/user)
 	. = ..()
 	if(. && !QDELETED(src) && !shock(user, 70))
```

Comment:
I think it's good without inverting, from what I can see from other code around it.

Beyond that, I did change the logic of this around a bit (since it was often getting destroyed in the parent call before anything else).